src/allotropy/parsers/tecan_magellan/tecan_magellan_structure.py

```diff
@@ -0,0 +1,204 @@
+        equipment_serial_number = raw_serial_number.split(": ")[1]
+
+        return MagellanMetadata(
+            measurement_time=measurement_time,
+            analytical_method_identifier=analytical_method_identifier,
+            experimental_data_identifier=experimental_data_identifier,
+            analyst=analyst,
+            device_identifier=device_identifier,
+            equipment_serial_number=equipment_serial_number,
+            measurements_settings=cls._get_measurements_settings(reader),
+        )
+
+    @classmethod
+    def parse_measurement_time(cls, raw_datetime: str) -> str:
+        if match := re.match(constants.MEASUREMENT_TIME_REGEX, raw_datetime):
+            return f"{match.groups()[0]} {match.groups()[1]}"
+        msg = "Unable to get measurement time from {raw_datetime}"
+        raise AllotropeConversionError(msg)
+
+    @classmethod
+    def _get_identifier_line(cls, reader: LinesReader, match_pat: str) -> str | None:
+        if lines := list(reader.pop_while(match_pat)):
+            return lines[-1]
+        return None
+
+    @classmethod
+    def _get_measurements_settings(
+        cls, reader: LinesReader
+    ) -> dict[str, MeasurementSettings]:
+        settings_lines: list[list[str]] = []
+        settings: dict[str, MeasurementSettings] = {}
+        label_and_temps = []
+        temp_line_re = r"Meas. temperature: (.+): ([\d\.]+) °C"
+
+        while not re.match(temp_line_re, (first_line := assert_not_none(reader.get()))):
+            if match := re.match(r"Measurement \d.", first_line):
+                reader.pop()
+            settings_lines.append(list(reader.pop_until_inclusive(r"Unit:.+")))
+
+        while match := re.match(temp_line_re, assert_not_none(reader.pop())):
+            label_and_temps.append(match.groups())
+
+        if len(label_and_temps) != len(settings_lines):
+            msg = "Number of temperatures does not match number of measurements."
+            raise AllotropeConversionError(msg)
+
+        for idx, lines in enumerate(settings_lines):
+            label, temperature = label_and_temps[idx]
+            settings[str(label)] = MeasurementSettings.create(
+                lines, try_float(str(temperature), "Measurement temperature")
+            )
+
+        return settings
+
+
+def create_metadata(data: MagellanMetadata, file_path: str) -> Metadata:
+    path = Path(file_path)
+    return Metadata(
+        unc_path=file_path,
+        file_name=path.name,
+        asm_file_identifier=path.with_suffix(".json").name,
+        device_identifier=data.device_identifier,
+        model_number=NOT_APPLICABLE,
+        data_system_instance_id=NOT_APPLICABLE,
+        software_name=constants.SOFTWARE_NAME,
+        product_manufacturer=constants.PRODUCT_MANUFACTURER,
+        equipment_serial_number=data.equipment_serial_number,
+    )
+
+
+def create_measurement_groups(
+    data: pd.DataFrame, metadata: MagellanMetadata
+) -> list[MeasurementGroup]:
+    return [
```

Comment:
You can use `functools.partial` to bind the common aguments of the function and use map_rows. This might not be exactly correct, but I think it's pretty close:

```suggestion
    from functools import partial
    return [group for group in map_rows(data, partial(_create_measurement_group, metadata=metadata, well_count=len(data))) if group]
```

Reply:
Thanks for the suggestion!

I actually ended up adding logic for capturing missing measurements to error documents and got rid of the condition at the end.